openjdk.test.load/src/test.load/net/adoptopenjdk/stf/MauveMultiThreadLoadTestWithTraceOption.java

```diff
@@ -0,0 +1,95 @@
+* limitations under the License.
+*******************************************************************************/
+
+package net.adoptopenjdk.stf;
+
+import net.adoptopenjdk.loadTest.InventoryData;
+import net.adoptopenjdk.stf.codeGeneration.Stage;
+import net.adoptopenjdk.stf.environment.FileRef;
+import net.adoptopenjdk.stf.environment.JavaVersion;
+import net.adoptopenjdk.stf.extensions.core.StfCoreExtension;
+import net.adoptopenjdk.stf.extensions.core.StfCoreExtension.Echo;
+import net.adoptopenjdk.stf.processes.ExpectedOutcome;
+import net.adoptopenjdk.stf.plugin.interfaces.StfPluginInterface;
+import net.adoptopenjdk.stf.processes.definitions.LoadTestProcessDefinition;
+import net.adoptopenjdk.stf.runner.modes.HelpTextGenerator;
+
+public class MauveMultiThreadLoadTestWithTraceOption implements StfPluginInterface {
+	public void help(HelpTextGenerator help) throws StfException {
+		help.outputSection("MauveMultiThreadLoadTest");
+		help.outputText("The MauveMultiThreadLoadTest runs a subset of tests from the GNU mauve project. "
+				+ "All tests are thread safe.");
+	}
+
+	public void pluginInit(StfCoreExtension stf) throws StfException {
+	}
+
+	public void setUp(StfCoreExtension test) throws StfException {
+	}
+
+	public void execute(StfCoreExtension test) throws StfException {
+
+		FileRef mauveJar = test.env().findPrereqFile("mauve/mauve.jar");
+
+		if (!mauveJar.asJavaFile().exists()) {
+			throw new StfException("Application jar file does not exist: " + mauveJar.getSpec());
+		}
+		
+		JavaVersion jvm = test.env().primaryJvm();
+
+		String inventoryFile = "/openjdk.test.load/config/inventories/mauve/mauve_multiThread.xml";
+		
+		int numMauveTests = InventoryData.getNumberOfTests(test, inventoryFile);
+		int cpuCount = Runtime.getRuntime().availableProcessors();
+
+		/*
+		 * Three Mauve tests require these add-opens options to run because modularity Java (9+) 
+		 * denies them the JCL module/package access they need to run to completion.
+		 */
+		String[] modularityOptions = null;
+		if (jvm.getJavaVersion() >= 9) { 
+			modularityOptions = new String[] {
+				"--add-opens=java.base/java.util=ALL-UNNAMED",
+				"--add-opens=java.base/java.lang=ALL-UNNAMED",
+				"--add-opens=java.base/java.io=ALL-UNNAMED"
+			}; 
+		}
+		
+		int multiplier = 1000; 
+		
+		// If special JIT modes are used, the test runs much slower, so we need to use a reduced amount of load
+		if (test.isJavaArgPresent(Stage.EXECUTE, "-Xjit:count=0") || 
+				test.isJavaArgPresent(Stage.EXECUTE, "-Xjit:count=0,optlevel=warm,gcOnResolve,rtResolve") ||
+				test.isJavaArgPresent(Stage.EXECUTE, "-Xjit:enableOSR,enableOSROnGuardFailure,count=1,disableAsyncCompilation")) {
+				multiplier = 250;
+		}
+
+		LoadTestProcessDefinition loadTestInvocation = test.createLoadTestSpecification()
+				.addJvmOption(modularityOptions)
+				.addJvmOption("\"-Xtrace:iprint=mt,trigger=method{java/util/Locale.setDefault*,jstacktrace}\"")
```

Comment:
Are you sure the quotes are required in the code? They are required on the command line to avoid problems with the shell, but shouldn't have those problems in code.

Reply:
i.e. how do we know the option is working properly?

Reply:
The options are being passed onto the test's command line: 

```
18:31:19  STF 15:31:18.792 - Running command: /home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image/bin/java "-Xtrace:iprint=mt,trigger=method{java/util/Locale.setDefault*,jstacktrace}" -Xcompressedrefs -Xjit:enableOSR,enableOSROnGuardFailure,count=1,disableAsyncCompilation -Xgcpolicy:gencon "-Xtrace:iprint=mt,trigger=method{java/util/Locale.setDefault*,jstacktrace}" 
```
https://hyc-runtimes-jenkins.swg-devops.com/view/Test_grinder/job/Grinder/8552/consoleFull

Reply:
If quotes aren't used, STF rejects the -Xtrace option: https://hyc-runtimes-jenkins.swg-devops.com/view/Test_grinder/job/Grinder/8549/consoleFull